rest/src/main/kotlin/builder/component/SelectMenuBuilder.kt

```diff
@@ -69,3 +73,45 @@ public class SelectMenuBuilder(
     }
 
 }
+
+@KordDsl
+public class StringSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.StringSelect
+}
+
+@KordDsl
+public class UserSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.UserSelect
+}
+
+@KordDsl
+public class RoleSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.RoleSelect
+}
+
+@KordDsl
+public class MentionableSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.MentionableSelect
+}
+
+@KordDsl
+public class ChannelSelectBuilder(customId: String) : SelectMenuBuilder(customId) {
+    override val type: ComponentType = ComponentType.ChannelSelect
+
+    private var _channelTypes: Optional<List<ChannelType>> = Optional.Missing()
+
+    public var channelTypes: List<ChannelType>? by ::_channelTypes.delegate()
```

Comment:
I suggest using `MutableList` here - it's more flexible and we use it in most of the other builders.